### PR TITLE
make systemd timers configurable

### LINKED ISSUE #107: Systemd timers should be configurable
## Use Case
Currently the timers are hard coded. In larger environments it can make sense to choose different values

## Describe the Solution You Would Like
Move the timers to parameters and data to make them configurable.

## Describe Alternatives You've Considered
none considered

## Additional Context
nope

### PR DESCRIPTION
**Pull request by SimonHoenscheid**
closes #107

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -8,23 +10,30 @@
...
-  Boolean $manage_postgresql_settings  = true,
-  Boolean $manage_table_settings       = false,
-  String  $install_dir                 = '/opt/puppetlabs/pe_databases',
-  String  $scripts_dir                 = "${install_dir}/scripts"
+  Boolean $manage_database_maintenance           = true,
+  Boolean $disable_maintenance                   = false,
+  Boolean $manage_postgresql_settings            = true,
+  Boolean $manage_table_settings                 = false,
+  String  $install_dir                           = '/opt/puppetlabs/pe_databases',
+  String  $scripts_dir                           = "${install_dir}/scripts",
```
Comment (bastelfreak) [inline manifests/init.pp:24]: I suggest we can assume it's an absolute path:
```suggestion
  Stdlib::Absolutepath  $scripts_dir             = "${install_dir}/scripts",
```

File: manifests/init.pp [hunk 2]
```diff
@@ -8,23 +10,30 @@
...
-  Boolean $disable_maintenance         = lookup('pe_databases::disable_maintenance', {'default_value' => false}),
-  Boolean $manage_postgresql_settings  = true,
-  Boolean $manage_table_settings       = false,
-  String  $install_dir                 = '/opt/puppetlabs/pe_databases',
-  String  $scripts_dir                 = "${install_dir}/scripts"
+  Boolean $manage_database_maintenance           = true,
+  Boolean $disable_maintenance                   = false,
+  Boolean $manage_postgresql_settings            = true,
+  Boolean $manage_table_settings                 = false,
+  String  $install_dir                           = '/opt/puppetlabs/pe_databases',
```
Comment (SimonHoenscheid) [inline manifests/init.pp:23]:
```suggestion
  Stdlib::Absolutepath $install_dir                           = '/opt/puppetlabs/pe_databases',
```

File: manifests/init.pp [hunk 3]
```diff
@@ -10,13 +10,23 @@
...
-  Boolean $disable_maintenance         = false,
-  Boolean $manage_postgresql_settings  = true,
-  Boolean $manage_table_settings       = false,
-  String  $install_dir                 = '/opt/puppetlabs/pe_databases',
-  String  $scripts_dir                 = "${install_dir}/scripts"
+  Boolean $manage_database_maintenance           = true,
+  Boolean $disable_maintenance                   = false,
+  Boolean $manage_postgresql_settings            = true,
+  Boolean $manage_table_settings                 = false,
+  Stdlib::Absolutepath $install_dir              = '/opt/puppetlabs/pe_databases',
```
Comment (MartyEwings) [inline manifests/init.pp:23]: This is a similar issue, as stdlib is not shipped with PE we cant add it to this module unfortunately

**Review by MartyEwings — CHANGES_REQUESTED** [review]
Due to the nature of this module being shipped with PE, even using a supported module such as stdlib is not something we can do within the confines of this project.

you may just have to declare this as a string

**Comment by bastelfreak** [comment]
@MartyEwings why can't you vendor stdlib or other modules? Puppet Inc and Vox Pupuli are working on best practice guidelines since years. a few of the topics are:
* make datatypes as strict as possible
* reuse existing datatypes
* rely on puppet/systemd to unify unit management and reduce exec resources to reload systemd

I'm always a bit afraid that new users first check modules from Puppet and assume the used code style is the current best practice. pe_databases would be a bad example here. Would'nt it make more sense to rely on external modules and use them in PE?

**Comment by MartyEwings** [comment]
@bastelfreak I work within the confines of a contributor to PE from the customer success organization, The overarching product team decides the standards of what is shipped with the Product. Im not sure what the specifics of the stdlib outside of PE discussion was, but there are a few good reasons I'm told.

With regards to dependencies, Everything my group puts in must not interfere with an offline installation, and the more dependencies I have the more my team must maintain and test and package.

In order to get the Product functions we require in, we have built everything for native Puppet Enterprise. Its not ideal, but its the practical way to contribute these functions we consider to be critical

**Review by m0dular — APPROVED** [review]
:+1

**Review by MartyEwings — APPROVED** [review]

**Pull request merged**